MekHQ/src/mekhq/gui/StratconTab.java

```diff
@@ -0,0 +1,286 @@
+    }
+    
+    /**
+     * Refreshes the list of tracks
+     */
+    private void repopulateTrackList() {
+        TrackDropdownItem currentTDI = (TrackDropdownItem) cboCurrentTrack.getSelectedItem();
+        cboCurrentTrack.removeAllItems();
+        
+        // track dropdown is populated with all tracks across all active contracts
+        for (Contract contract : getCampaignGui().getCampaign().getActiveContracts()) {
+            if ((contract instanceof AtBContract) && 
+            		contract.isActiveOn(getCampaignGui().getCampaign().getLocalDate()) && 
+            		(((AtBContract) contract).getStratconCampaignState() != null)) {
+                for (StratconTrackState track : ((AtBContract) contract).getStratconCampaignState().getTracks()) {
+                    TrackDropdownItem tdi = new TrackDropdownItem((AtBContract) contract, track);
+                    cboCurrentTrack.addItem(tdi);
+                    
+                    if ((currentTDI != null) && currentTDI.equals(tdi)) {
+                        currentTDI = tdi;
+                        cboCurrentTrack.setSelectedItem(tdi);
+                    } else if (currentTDI == null) {
+                        currentTDI = tdi;
+                        cboCurrentTrack.setSelectedItem(tdi);
+                    }
+                }
+            }
+        }
+        
+        if ((cboCurrentTrack.getItemCount() > 0) && (currentTDI != null) && (currentTDI.contract != null)) {
+            TrackDropdownItem selectedTrack = (TrackDropdownItem) cboCurrentTrack.getSelectedItem();
+            
+            stratconPanel.selectTrack(selectedTrack.contract.getStratconCampaignState(), currentTDI.track);
+            stratconPanel.setVisible(true);
+        } else {
+            infoPanelText.setText("No active campaign tracks");
+            stratconPanel.setVisible(false);
+        }
+    }
+    
+    @Subscribe
+    public void handleNewDay(NewDayEvent ev) {
+        repopulateTrackList();
+        updateCampaignState();
+    }
+    
+    @Subscribe
+    public void handle(MissionRemovedEvent ev) {
+        repopulateTrackList();
+        updateCampaignState();
+    }
+
+    @Subscribe
+    public void handle(MissionCompletedEvent ev) {
+        repopulateTrackList();
+        updateCampaignState();
+    }
+    
+    /**
+     * Data structure to hold necessary information about a track drop down item.
+     * @author NickAragua
+     *
+     */
+    private class TrackDropdownItem {
+        AtBContract contract;
+        StratconTrackState track;
+        
+        public TrackDropdownItem(AtBContract contract, StratconTrackState track) {
+            this.contract = contract;
+            this.track = track;
+        }
+        
+        @Override
+        public String toString() {
+            return String.format("%s - %s", contract.getName(), track.getDisplayableName());
+        }
+        
+        @Override
+        public boolean equals(Object other) {
```

Comment:
Don't forget:
```java
public int hashCode() {
    return Objects.hash(this.contract, this.track);
}
```

Reply:
What's the point? Neither the AtBContract nor the StratconTrackState implement hash. Do objects come with a meaningful default hash function?

Reply:
You must always override hashCode for any file that overrides equals, it's a Java contract.

Reply:
The contract is specifically that if two objects return true for `equals` they must have the same hash code.